packages/client/base/src/services/embed/v3/crossmintEmbeddedCheckoutV3Service.ts

```diff
@@ -0,0 +1,52 @@
+import { CrossmintEmbeddedCheckoutV3Props } from "@/types/embed/v3/CrossmintEmbeddedCheckoutV3Props";
+import { embeddedCheckoutV3IncomingEvents, embeddedCheckoutV3OutgoingEvents } from "@/types/embed/v3/events";
+
+import { IFrameWindow } from "@crossmint/client-sdk-window";
+import { CrossmintClientSideApiClient } from "@crossmint/common-sdk-base";
+
+export type CrossmintEmbeddedCheckoutV3ServiceProps = { apiClient: CrossmintClientSideApiClient };
+
+export function crossmintEmbeddedCheckoutV3Service({ apiClient }: CrossmintEmbeddedCheckoutV3ServiceProps) {
+    function getIFrameUrl(props: CrossmintEmbeddedCheckoutV3Props) {
+        const urlWithPath = apiClient.buildUrl("/sdk/2024-03-05/embedded-checkout");
+        const queryParams = new URLSearchParams();
+
+        let key: keyof CrossmintEmbeddedCheckoutV3Props;
+        for (key in props) {
+            const value = props[key] as unknown;
+
+            if (!value || typeof value === "function") {
+                continue;
+            }
+            if (typeof value === "object") {
+                queryParams.append(
+                    key,
+                    JSON.stringify(value, (key, val) => (typeof val === "function" ? "function" : val))
+                );
+            } else if (typeof value === "string") {
+                if (value === "undefined") {
+                    continue;
+                }
+                queryParams.append(key, value);
+            } else if (["boolean", "number"].includes(typeof value)) {
+                queryParams.append(key, value.toString());
+            }
+        }
```

Comment:
nearly exactly the same as https://github.com/Crossmint/crossmint-sdk/blob/main/packages/client/base/src/services/embed/crossmintIFrameService.ts

i think it will need to be changed a bit later tho, and dont want to change existing services, so fine with duplicating for now